src/prefect/engine/flow_runner.py

```diff
@@ -461,26 +460,28 @@ def get_flow_run_state(
 
                     ## wait on upstream states to determine the width of the pipeline
                     ## this is the key to depth-first execution
-                    upstream_states.update(
-                        executor.wait(
-                            {e: state for e, state in upstream_states.items()}
-                        )
+                    upstream_states = executor.wait(
+                        {e: state for e, state in upstream_states.items()}
+                    )
+                    _set_upstream_states_map_states(
```

Comment:
In all of these `_set_upstream_states_map_states ` calls, how do we know that all futures in `.map_states` have been resolved to actual `State` objects?

Reply:
I was mistaken about where things were being awaited, reverted.